allennlp/common/testing/model_test_case.py

```diff
@@ -92,18 +86,29 @@ def ensure_model_can_train_save_and_load(self, param_file: str, tolerance: float
         for key in model_predictions.keys():
             self.assert_fields_equal(model_predictions[key],
                                      loaded_model_predictions[key],
-                                     tolerance=1e-4,
+                                     tolerance=tolerance,
                                      name=key)
 
         return model, loaded_model
 
-    @staticmethod
-    def assert_fields_equal(field1, field2, tolerance: float = 1e-6, name: str = None) -> None:
+    def assert_fields_equal(self, field1, field2, tolerance: float = 1e-6, name: str = None) -> None:
         if isinstance(field1, torch.autograd.Variable):
             assert_allclose(field1.data.cpu().numpy(),
                             field2.data.cpu().numpy(),
                             rtol=tolerance,
                             err_msg=name)
+        elif isinstance(field1, dict):
+            for key in field1:
```

Comment:
should you also check that they have the same keys? right now this would pass if field2 had additional keys

Reply:
Yep, great point, thanks.